### PR TITLE
Unsubscribe only if there is a subscription

### PR DESCRIPTION
**Pull request by dotbear**

### CODE REVIEW

File: src/StateProvider.js [hunk 1]
```diff
@@ -121,13 +121,19 @@ export default class StateProvider {
 
       static displayName = `ProvideState(${getDisplayName(Component)})`;
 
+      isSubscribed() {
+        return typeof this.unsubscribe === 'function';
```
Comment (SomeHats) [inline src/StateProvider.js]: Can we change the name of this or just inline the check? this doesn't reflect whether or not the subscription is active - `this.unsubscribe()` is still a function after it's been called

Comment (dotbear) [inline src/StateProvider.js]: Took it from here and thought it was nice: https://github.com/reactjs/react-redux/commit/bda582b728514446594bcef3c2f39ce52dba02d9

I can easily inline it :)

Comment (SomeHats) [inline src/StateProvider.js]: well it's wrong there too imo :P inlining would be good :)

**Comment by jgwmaxwell** [comment]
Approved

[Approved with PullApprove](https://pullapprove.com/musicglue/react-relax/pull-request/1/?utm_source=github-pr&utm_medium=comment-badge&utm_campaign=musicglue/react-relax)

**Comment by dotbear** [comment]
Approved

[Approved with PullApprove](https://pullapprove.com/musicglue/react-relax/pull-request/1/?utm_source=github-pr&utm_medium=comment-badge&utm_campaign=musicglue/react-relax)

**Pull request merged**